SharpNoise.Tests/ModuleTests/DiscreteTests.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SharpNoise.Modules;
+
+namespace SharpNoise.Tests.ModuleTests
+{
+    [TestClass]
+    public class DiscreteTests
+    {
+        Random random;
+
+        const Byte VALUES = 4;
+
+        public void DiscreteMainTestCode(Module source, Byte Count = VALUES)
+        {
+            Assert.IsNotNull(source);
+
+            Discrete module = new Discrete() { Source0 = source, Values = Count };
+
+            random = new Random((int)DateTime.Now.Ticks & (0x0000FFFF));
+
+            double sourceValue, checkvalue;
+            double x, y, z;
+
+            for (Byte count = 0; count < 50; count++)

```

Comment:
You should  usually never write loop like that. It will perform slower and generate more assembly code than a loop with 'int' variable. The reason is that almost all current architectures operate on 32 bit registers (I think even 64bit architectures handle 32bit values directly), but operating on 8 or 16 bit values requires extra effort.

As a good rule of thumb, all integer number local variables and method parameters should be int (or uint), unless you explicitly need a smaller or larger type.